frontend/src/components/entities/create-entity-dialog.tsx

```diff
@@ -0,0 +1,185 @@
+"use client"
+
+import { zodResolver } from "@hookform/resolvers/zod"
+import { useState } from "react"
+import { useForm } from "react-hook-form"
+import { z } from "zod"
+import { IconPicker } from "@/components/form/icon-picker"
+import { Button } from "@/components/ui/button"
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog"
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form"
+import { Input } from "@/components/ui/input"
+import { Textarea } from "@/components/ui/textarea"
+
+const createEntitySchema = z.object({
+  key: z
+    .string()
+    .min(1, "Key is required")
+    .regex(
+      /^[a-z][a-z0-9_]*$/,
+      "Key must start with a letter, be lowercase, and contain only letters, numbers, and underscores"
+    ),
+  display_name: z.string().min(1, "Display name is required"),
+  description: z.string().optional(),
+  icon: z.string().optional(),
+})
+
+type CreateEntityFormData = z.infer<typeof createEntitySchema>
+
+interface CreateEntityDialogProps {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+  onSubmit: (data: CreateEntityFormData) => Promise<void>
+}
+
+export function CreateEntityDialog({
+  open,
+  onOpenChange,
+  onSubmit,
+}: CreateEntityDialogProps) {
+  const [isSubmitting, setIsSubmitting] = useState(false)
+
+  const form = useForm<CreateEntityFormData>({
+    resolver: zodResolver(createEntitySchema),
+    defaultValues: {
+      key: "",
+      display_name: "",
+      description: "",
+      icon: "",
+    },
+  })
+
+  const handleSubmit = async (data: CreateEntityFormData) => {
+    setIsSubmitting(true)
+    try {
+      await onSubmit(data)
+      form.reset()
+      onOpenChange(false)
+    } catch (error) {
+      console.error("Failed to create entity:", error)
+    } finally {
+      setIsSubmitting(false)
```

Comment:
prefer to tie this to server side state (react query) instead of client side state